ledger-service/http-json/src/main/scala/com/digitalasset/http/HttpService.scala

```diff
@@ -284,4 +287,14 @@ object HttpService extends StrictLogging {
         case NonFatal(e) =>
           \/.left(Error(s"Cannot connect to the ledger server, error: ${e.description}"))
       }
+
+  private def updatePortFile(
+      file: Path,
+      binding: akka.http.scaladsl.Http.ServerBinding): Error \/ Unit = {
+    import scala.collection.JavaConverters._
+    val lines: java.lang.Iterable[String] = List(binding.localAddress.getPort.toString).asJava
+    \/.fromTryCatchNonFatal(Files.write(file, lines))
```

Comment:
We should fail if the file exists:

```suggestion
    \/.fromTryCatchNonFatal(Files.write(file, lines, StandardOpenOption.CREATE_NEW))
```

Additionally, we should arrange to delete the file, then, on exit.

Reply:
I copied what sandbox does.... sandbox does not check for existing files.